speclang_modern.tex

```diff
@@ -3026,14 +3044,16 @@ \subsection{Semantics}
   Here is a longer example, the famous Dijkstra's Dutch flag algorithm.\\
   \listinginput{1}{flag.c}
 \end{example}
+\marginpar{Why is the invariant not strong in the Dutch flag example? Also, perhaps such an invariant should be automatically generated for an enum. }
```

Comment:
The invariant could indeed be made `strong`. It is certainly possible to provide some syntactic sugar to generate such an invariant for an `enum`, but this should not be systematic: there are use cases of the form:
```c
enum flags = { flag_0 = 0x1, flag_1 = 0x2, flag_2 = 0x4, ... };
...
enum flags f = flag_0 | flag_1; 
...
if (f & flag_3) { ... }
```

where it is normal to have values that are not in the enumeration.